vital/types/bounding_box.h

```diff
@@ -95,6 +95,22 @@ class bounding_box
     m_bbox =  Eigen::AlignedBox< T, 2 >( ul, lr );
   }
 
+  /**
+   * @brief Create default (invalid) box.
+   */
+  bounding_box() : bounding_box( -1, -1, -1, -1 )
```

Comment:
Are you sure? If I am reading correctly, this creates a box with TL = (-1, -1), W=0, H=0, which is *valid*.

I think you want `: bounding_box{ 0, 0, -1, -1 }`?

Reply:
This ctor I am using is setting the upper and lower left/right values (not width and height)
W, H is computed by eigen
Even if I did call what you suggest (using 0 width), I would get an eigen box of all -1's

We kind of talked about what is an invalid box before, and you brought up the point that
 _Can't predictive boxes have corners less than 0? (I think the "safe" way to implement this, assuming Eigen lets you, is to define "invalid" as having a negative width and/or height. I want to say that's how e.g. QRect[F] is implemented...)_

Would any box that has all the same values for each point be invalid?

I am not sure

Reply:
> This ctor I am using is setting the upper and lower left/right values

That is consistent with my reading of the code. Since `is_valid` was previously true for boxes with non-negative width and height, I suggested instead creating a box that would have negative width and height, by means of the bottom and right coordinates being less than top and left, respectively. (It looks like this is the same way Eigen represents "empty".)

Alternatively, AlignedBox actually has an initializing default ctor; you could just use that, and further have `is_valid` just return `!m_bbox.isEmpty()`.

Actually, this made me think of something else; have you considered adding a `reset` method?

Reply:
I like this ideal!